### PR TITLE
(WIP) Mpl backend

### PR DESCRIPTION
**Pull request by alejoe91**

### CODE REVIEW

**Comment by alejoe91** [comment]
@samuelgarcia I like the idea of having a basewidget that handles the figure / ax

File: setup.py [hunk 1]
```diff
@@ -15,16 +15,8 @@
...
-        'ipywidgets'
-        # 'scipy',
-        # 'ipython',
-        # 'vdom',
-        # 'ipywidgets',
-        # 'matplotlib',
-        # 'spikeextractors',
-        # 'pandas',
-        # 'pillow'
+        'spiketoolkit',
```
Comment (samuelgarcia) [inline setup.py:18]: This is an important choice : does spiketoolit depend on spikewidget or the reverse.
If spikewidget depend on spiketoolkit, then we will have only the functional approach for plotting.
We will use the method approach like SpikeComparison.plot_confusion_matrix() and keep only
spikewidget.plot_confusion_matrix(spikecomparison)

Comment (alejoe91) [reply setup.py:18]: I think it's better to have spikewidget depending on spiketoolkit. This is helpful (for example in the plot_features I use st to compute pca and it's very short).

I think it's time to remove the plot from the sorting comparison eheheh

Comment (samuelgarcia) [reply setup.py:18]: I am not ready :(
Wait a bit more.

**Pull request merged**